### PR TITLE
ARROW-137 Remove workarounds for extension types in test suite

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: bindings/python/test/test_arrow.py [hunk 1]
```diff
@@ -244,27 +244,43 @@ def test_pymongo_error(self):
             exc.exception.details.keys(), {"nInserted", "writeConcernErrors", "writeErrors"}
         )
 
-    def _create_data(self):
+    def _create_data(self, block=False, nested=True):
```
Comment (blink1073) [inline bindings/python/test/test_arrow.py:247]: What does "block" mean?  It looks like this is only ever called with `block=True`.  When would we not want to include the extension types?

**Review by blink1073 — CHANGES_REQUESTED** [review]
There is also a `.DS_Store` file committed.

File: bindings/python/test/utils.py [hunk 2]
```diff
@@ -145,7 +145,10 @@ def setUp(self):
 
     def assertType(self, obj1, arrow_type):
         if isinstance(obj1, pyarrow.ChunkedArray):
-            if "storage_type" in dir(arrow_type):
+            if "storage_type" in dir(arrow_type) and arrow_type not in (
+                ObjectIdType(),
```
Comment (blink1073) [inline bindings/python/test/utils.py:149]: Perhaps we can check for an extension type here so we don't have to manually update?

**Comment by ShaneHarvey** [comment]
Should we close this PR now based on the extra work required?

**Pull request closed without merging**